### PR TITLE
Rewrite #720 with pytest

### PR DESCRIPTION
**Pull request by slsh1o**
Relates: #707 
Close: #720

### CODE REVIEW

File: test/TestImportWithMalformed.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    rulesdir = os.path.join('lib', 'ansiblelint', 'rules')
+    return RulesCollection([rulesdir])
+
+
+@pytest.fixture
+def runner(play_file_path, rules):
+    return Runner(rules, play_file_path, [], [], [])
+
+
+
```
Comment (webknjaz) [inline test/TestImportWithMalformed.py:40]:
```suggestion
```

File: test/TestImportWithMalformed.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import pytest
+
+from ansiblelint import Runner, RulesCollection
+
+
+PlayFile = namedtuple('PlayFile', ['name', 'content'])
+
+
+IMPORT_TASKS_MAIN = PlayFile('import-tasks-main.yml', '''
+- oops this is invalid
```
Comment (webknjaz) [inline test/TestImportWithMalformed.py:13]: I think I'd add another playbook but with `- ping:` instead and add that to parametrize

File: test/TestImportWithMalformed.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+@pytest.fixture
+def play_file_path(tmp_path):
+    p = tmp_path / 'playbook.yml'
+    return str(p)
+
+
+@pytest.fixture
+def rules():
+    rulesdir = os.path.join('lib', 'ansiblelint', 'rules')
```
Comment (webknjaz) [inline test/TestImportWithMalformed.py:31]: Let's use pathlib instead
```suggestion
    rulesdir = str(Path('lib') / 'ansiblelint' / 'rules')
```

File: test/TestImportWithMalformed.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+from collections import namedtuple
+import os
```
Comment (webknjaz) [inline test/TestImportWithMalformed.py:2]:
```suggestion
from pathlib import Path
```

**Comment by slsh1o** [comment]
I believe I made it wrong with `-ping:` or we need something else because this _ping_ doesn't fit assert msg.
As we fix it I'll be able to add xfail to the parametrize.

**Comment by webknjaz** [comment]
@slsh1o what's left now is to add an xfail marker to the parameter that causes `AttributeError`.

File: test/TestImportWithMalformed.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+
+@pytest.mark.parametrize(
+    'play_files',
+    [
+        pytest.param([IMPORT_SHELL_PIP, PLAY_IMPORT_TASKS], id='Import shell w/ pip'),
+        pytest.param(
+            [IMPORT_TASKS_MAIN, PLAY_IMPORT_TASKS],
+            id='import_tasks w/ malformed import',
+            marks=pytest.mark.xfail(raises=AttributeError)
```
Comment (webknjaz) [inline test/TestImportWithMalformed.py:60]: Nitpick: normally it's a good practice to also specify a `reason=` argument pointing the issue.

**Pull request merged**